File: docs/snippets/all/archetypes/line_strips3d_simple.py
"""Log a simple line strip."""

import dalaran as dl

dl.init("dalaran_example_line_strip3d", spawn=True)

points = [
    [0, 0, 0],
    [0, 0, 1],
    [1, 0, 0],
    [1, 0, 1],
    [1, 1, 0],
    [1, 1, 1],
    [0, 1, 0],
    [0, 1, 1],
]

dl.log("strip", dl.LineStrips3D([points]))

File: docs/snippets/all/archetypes/line_strips3d_simple.rs
//! Log a simple line strip.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new("dalaran_example_line_strip3d")
        .spawn()?;

    let points = [
        [0., 0., 0.],
        [0., 0., 1.],
        [1., 0., 0.],
        [1., 0., 1.],
        [1., 1., 0.],
        [1., 1., 1.],
        [0., 1., 0.],
        [0., 1., 1.],
    ];
    rec.log("strip", &dalaran::LineStrips3D::new([points]))?;

    Ok(())
}
